src/services/dog.js

```diff
@@ -0,0 +1,39 @@
+import { checkError, client } from './client';
+
+export async function fetchDogs() {
+  const resp = await fetch(`${process.env.REACT_APP_SUPABASE_URL}/rest/v1/dogs?`, {
```

Comment:
I would try to keep this consistent and only use the supabase client for your requests